controllers/chat/ChatController.js

```diff
@@ -91,10 +91,15 @@ module.exports = {
   getChannels: async (req, res) => {
     const client = new StreamChat(process.env.API_KEY, process.env.SECRET);
     try {
+      let {last_fetch_date = null} = req.query;
+      let filter = {members: {$in: [req.userId]}};
+      if (last_fetch_date) {
+        last_fetch_date = new Date(last_fetch_date);
+        last_fetch_date = last_fetch_date.toISOString();
+        filter = {members: {$in: [req.userId]}, last_message_at: {$gte: last_fetch_date}};
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is not handling the case when `last_fetch_date` is not a valid date string. This could lead to unexpected behavior if an invalid date string is passed as `last_fetch_date`. It's better to validate the date string before converting it to a Date object.

```diff
-       let {last_fetch_date = null} = req.query;
-       let filter = {members: {$in: [req.userId]}};
-       if (last_fetch_date) {
-         last_fetch_date = new Date(last_fetch_date);
-         last_fetch_date = last_fetch_date.toISOString();
-         filter = {members: {$in: [req.userId]}, last_message_at: {$gte: last_fetch_date}};
-       }
+       const {last_fetch_date = null} = req.query;
+       const filter = {members: {$in: [req.userId]}};
+       if (last_fetch_date) {
+         const parsedDate = new Date(last_fetch_date);
+         if (!isNaN(parsedDate)) {
+           const isoDate = parsedDate.toISOString();
+           filter.last_message_at = {$gte: isoDate};
+         } else {
+           return res.status(400).json({ error: 'Invalid last_fetch_date' });
+         }
+       }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->